azure/packages/test/end-to-end-tests/src/test/utils.ts

```diff
@@ -2,15 +2,19 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+import { IMember } from "fluid-framework";
+
 import { AzureMember, IAzureAudience } from "@fluidframework/azure-client";
 import { ISharedMap, IValueChanged } from "@fluidframework/map";
 
-export const waitForMyself = async (audience: IAzureAudience): Promise<AzureMember> => {
+export const waitForMyself = async (
+    audience: IAzureAudience,
+    userId: string,
+): Promise<AzureMember> => {
     return new Promise((resolve) => {
```

Comment:
If the user is already in the audience this will never resolve, consider something like this to make it more robust:
```suggestion
    const member = audience.getMember(userId);
    if (member !== undefined) {
        return member;
    }
    return new Promise((resolve) => {
```

Reply:
Good catch. This is valid scenario that can play out.